pandas/io/formats/format.py

```diff
@@ -1246,7 +1246,7 @@ def format_percentiles(percentiles):
             raise ValueError("percentiles should all be in the interval [0,1]")
 
     percentiles = 100 * percentiles
-    int_idx = (percentiles.astype(int) == percentiles)
+    int_idx = (abs(percentiles.astype(int) - percentiles) < 1e-4)
```

Comment:
What's the reasoning for using 1e-4? May make more sense just to leverage np.allclose

Reply:
Good insight - it makes code more readable. Instead of np.allclose, I've opted out for np.isclose, which returns array of booleans. Reasoning for that is that further down in the function that mask is used when determining how to format integers (where mask is True) and decimal numbers (where mask is False).

Another approach would be that if at least one percentile is decimal, all percentiles should be represented as decimals. This is currently not the case, for instance if percentiles to describe are: [0, 0.5, 0.334] the method will return ["0%", "50%", "33.4%"]. This approach would yield: ["0.0%", "50.0%", "33.4%"].

I wanted to change as little as possible, so I returned mask of booleans from this method, but if it fits your business logic to change to second approach, I can modify the code to that behavior also.

Reply:
Nice. This is fine as is - really just wanted to circumvent hard coding thresholds like before. Thanks for reviewing in detail!